### PR TITLE
Make deploying the web searcher plugin optional

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

The Web Searcher plugin requires a Bing resource, which may require elevated permission. This will fail the whole deployment, blocking developers from deploying the latest CC. 

Issues:
https://github.com/microsoft/chat-copilot/issues/498

### Description
This PR makes deploying the Web Searcher plugin optional. And by default, it's NOT deployed.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: scripts/deploy/README.md [hunk 1]
```diff
@@ -152,10 +152,13 @@ chmod +x ./deploy-webapi.sh
 
 # Deploy Hosted Plugins
 
-> **_NOTE:_** This step can be skipped if the previous Azure Resources creation step succeeded without errors. The `deployWebSearcherPackage = true` setting in main.bicep ensures that the WebSearcher is deployed.
+> **_NOTE:_** This step can be skipped if the required resources for the web searcher plugin are not deployed. The required resources include a Bing resource that may require an elevated permission, and an Azure Function. The required resources are NOT deployed by default. To deploy the required resources, use the `-DeployWebSearcherPlugin` or `--deploy-web-searcher-plugin` flag when running the **deploy-azure.ps1/deploy-azure.sh** script.
```
Comment (glahaye) [inline scripts/deploy/README.md:155]:
```suggestion
> **_NOTE:_** This step can be skipped if the required resources for the web search plugin are not deployed. The required resources include a Bing Search resource and an Azure Function. The required resources are NOT deployed by default. To deploy the required resources, use the `-DeployWebSearcherPlugin` or `--deploy-web-searcher-plugin` flag when running the **deploy-azure.ps1/deploy-azure.sh** script.
```

Don't mention elevated permissions.  That's just something specific to how our subscription is set up. Using another subscription doesn't have this problem.

File: scripts/deploy/main.bicep [hunk 2]
```diff
@@ -75,6 +75,13 @@ param memoryStore string = 'AzureCognitiveSearch'
 @description('Whether to deploy Azure Speech Services to enable input by voice')
 param deploySpeechServices bool = true
 
+/*
+The web searcher plugin requires a Bing resource.
+A Bing resource may require an elevated permission, which will block the deployment.
```
Comment (glahaye) [inline scripts/deploy/main.bicep:80]: This comment is incorrect and can be removed intirely.

File: scripts/deploy/main.bicep [hunk 3]
```diff
@@ -75,6 +75,13 @@ param memoryStore string = 'AzureCognitiveSearch'
...
+*/
+@description('Whether to the web searcher plugin, which requires a Bing resource')
```
Comment (glahaye) [inline scripts/deploy/main.bicep:82]:
```suggestion
@description('Whether to deploy the web search plugin, which requires a Bing resource')
```

File: scripts/deploy/main.bicep [hunk 4]
```diff
@@ -75,6 +75,13 @@ param memoryStore string = 'AzureCognitiveSearch'
...
+param deployWebSearcher bool = false
```
Comment (glahaye) [inline scripts/deploy/main.bicep:83]: deployWebSearch seems more grammatically correct to me.

File: scripts/deploy/main.bicep [hunk 5]
```diff
@@ -177,232 +184,235 @@ resource appServiceWebConfig 'Microsoft.Web/sites/config@2022-09-01' = {
...
+        {
+          name: 'Plugins:0:Name'
+          value: 'Klarna Shopping'
+        }
+        {
+          name: 'Plugins:0:ManifestDomain'
+          value: 'https://www.klarna.com'
+        }
+      ],
+      (deployWebSearcher) ? [
```
Comment (glahaye) [inline scripts/deploy/main.bicep:401]: Nice trick!

**Review by alliscode — DISMISSED** [review]

**Review by glahaye — APPROVED** [review]

**Pull request merged**